Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -476,12 +491,144 @@ void MemoryWidget::OnSearchAddress()
+                           combo_id == InputId::U8);
+
+  if (input_text.isEmpty())
+    return;
+
+  // Remove any spaces
+  if (combo_id != InputId::ASCII)
+    input_text.remove(QLatin1Char(' '));
+
   QFont font;
   QPalette palette;
+  QString hex_string;
+  bool good = true;
+
+  if (m_base_check->isChecked() && m_base_check->isEnabled())
+    input_text = QStringLiteral("0x") + input_text;
+
+  switch (combo_id)
+  {
+  case InputId::ASCII:
+  {
+    const QByteArray bytes = input_text.toLatin1();
+    hex_string = QString::fromLatin1(bytes.toHex());
+    break;
+  }
+  case InputId::FLOAT:
+  {
+    const float value_float = input_text.toFloat(&good);
+
+    if (good)
+    {
+      const u32 hex_out = Common::BitCast<u32>(value_float);
+      hex_string = QString::fromStdString(fmt::format("{:X}", hex_out));
+    }
+    break;
+  }
+  case InputId::DOUBLE:
+  {
+    const double value_double = input_text.toDouble(&good);
+
+    if (good)
+    {
+      const u64 hex_out = Common::BitCast<u64>(value_double);
+      hex_string = QString::fromStdString(fmt::format("{:X}", hex_out));
+    }
+    break;
+  }
+  case InputId::S8:
+  {
+    const short value = input_text.toShort(&good, 10);
+    if (good && value <= std::numeric_limits<signed char>::max() &&
+        value >= std::numeric_limits<signed char>::min())
+      hex_string = QString::fromStdString(fmt::sprintf("%02hhX", value));
+    else
+      good = false;
+    break;
+  }
+  case InputId::S16:
+  {
+    const short value = input_text.toShort(&good, 10);
+    if (good)
+      hex_string = QString::fromStdString(fmt::sprintf("%04hX", value));
+    break;
+  }
+  case InputId::S32:
+  {
+    const int value_int = input_text.toInt(&good, 10);
+    if (good)
+      hex_string = QString::fromStdString(fmt::sprintf("%08X", value_int));
+    break;
+  }
+  case InputId::U8:
+  {
+    const unsigned short value = input_text.toUShort(&good, 0);
+    if (good && (value & 0xFF00) == 0)
+      hex_string = QString::fromStdString(fmt::format("{:02X}", value));
+    else
+      good = false;
+    break;
+  }
+  case InputId::U16:
+  {
+    const unsigned short value = input_text.toUShort(&good, 0);
+    if (good)
+      hex_string = QString::fromStdString(fmt::format("{:04X}", value));
+    break;
+  }
+  case InputId::U32:
+  {
+    const u32 value = input_text.toUInt(&good, 0);
+    if (good)
+      hex_string = QString::fromStdString(fmt::format("{:08X}", value));
+    break;
+  }
+  case InputId::HEXSTR:
+  {
+    // Confirm it is only hex bytes
+    const QRegularExpression is_hex(QStringLiteral("^([0-9A-F]{2})*$"),
+                                    QRegularExpression::CaseInsensitiveOption);
+    const QRegularExpressionMatch match = is_hex.match(m_data_edit->text());
+
+    if (match.hasMatch())
+    {
+      good = true;
+      const QByteArray hbytes = QByteArray::fromHex(m_data_edit->text().toUtf8());
```

Comment:
The `m_data_edit->text()` here should be replaced with `input_text` so that spaces can be in the hex string.